File: backend/app/schemas/audit.py
"""审计日志 API 模型"""

from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class AuditLogItem(BaseModel):
    id: int
    username: Optional[str] = None
    action: str
    resource_type: str
    resource_id: Optional[int] = None
    project_id: Optional[int] = None
    detail: Optional[str] = None
    request_id: Optional[str] = None
    created_at: datetime

    model_config = {"from_attributes": True}


class AuditLogListResponse(BaseModel):
    items: List[AuditLogItem] = Field(default_factory=list)
    total: int = 0
    skip: int = 0
    limit: int = 50

File: frontend/src/types/audit.ts
/** 与 GET /audit-logs 对齐 */

export interface AuditLogItem {
  id: number;
  username: string | null;
  action: string;
  resource_type: string;
  resource_id: number | null;
  project_id: number | null;
  detail: string | null;
  request_id: string | null;
  created_at: string;
}

export interface AuditLogListResponse {
  items: AuditLogItem[];
  total: number;
  skip: number;
  limit: number;
}
